backend/app/notify/email.go

```diff
@@ -0,0 +1,114 @@
+package notify
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	log "github.com/go-pkgz/lgr"
+	"github.com/pkg/errors"
+	"gopkg.in/gomail.v2"
+)
+
+// Email implements notify.Destination for email
+type Email struct {
+	server    string
+	port      int
+	username  string
+	password  string
+	keepAlive time.Duration
+	ch        chan *gomail.Message
+}
+
+const emailConnectionKeepAlive = 30 * time.Second
+
+//NewEmail makes email object for notifications and run sending daemon
+func NewEmail(server string, port int, username string, password string, keepAlive time.Duration) (*Email, error) {
+
+	res := Email{
+		server:    server,
+		port:      port,
+		username:  username,
+		password:  password,
+		keepAlive: keepAlive,
+		ch:        make(chan *gomail.Message),
+	}
+
+	if res.keepAlive == 0 {
+		res.keepAlive = emailConnectionKeepAlive
+	}
+
+	log.Printf(
+		"[DEBUG] create new email notifier for server %s with user %s, keepalive=%s",
+		res.server, res.username, res.keepAlive)
+
+	// test connection before starting a daemon
+	tmpConn, err := gomail.NewDialer(res.server, res.port, res.username, res.password).Dial()
+	if err != nil {
+		return &res, errors.Errorf(
+			"[WARN] error connecting to '%s':%d with username '%s': %s",
+			res.server, res.port, res.username, err)
+	}
+	err = tmpConn.Close()
+	if err != nil {
+		return &res, errors.Errorf(
+			"[WARN] error closing connection to %s:%d: %s",
+			res.server, res.port, err)
+	}
+
+	go func() {
```

Comment:
I would extract this anon func to the normal/named, i.e. `func (e *Email) activate(ctx context.Context)`